framework/doc/content/source/problems/SolutionInvalidity.md

```diff
@@ -0,0 +1,43 @@
+# SolutionInvalidity
+
+## Description
+
+The [/SolutionInvalidity.md] object holds solution invalid warning information for MOOSE. With this object, you can mark a solution as "invalid" and output which object and how many times the warning occurs. An invalid solution means that the solution somehow does not satisfy requirements, such as a value being out of the bounds of a correlation.  Solutions are allowed to be invalid *during* the nonlinear solve, but they are not allowed to be invalid once it converges. A "converged" solution that is marked as invalid will cause MOOSE to behave as if the solution did NOT converge - including cutting back timesteps, etc.
+
+To declare a solution as "invalid", use the macro in the following code to mark it. The user can provide a message to describe the invalidity:
+
+!listing /test/src/materials/NonsafeMaterial.C  re=\s+if \(_input_diffusivity > _threshold\).*?\}
+
+!alert tip
+It is recommended to have a unique message for each invalidity, especially when you want to mark multiple types of invalid solutions within one object.
+
+If any solution invalidity is detected during the solve, a summary table of solution invalid warnings will be generated and reported at the end of each time step. An example of this output is shown below:
+
+```
+Solution Invalid Warnings:
+------------------------------------------------------------------------------------------------------
+|     Object      | Latest | Timestep | Total |                        Message                       |
+------------------------------------------------------------------------------------------------------
+| NonsafeMaterial |     16 |       64 |    64 | The diffusivity is greater than the threshold value! |
+| NonsafeMaterial |     16 |       64 |    64 | Extra invalid thing!                                 |
+------------------------------------------------------------------------------------------------------
```

Comment:
if we have 0 warnings during the timestep, does the warning get silenced?

Reply:
Yes, there will not be a summary table in the end.
![Screenshot 2023-02-12 at 8 42 05 PM](https://user-images.githubusercontent.com/118846840/218365449-880a61da-1155-4401-b116-8c66c43670bb.png)

Reply:
What if there s a warning in timestep 1 and no warning in timestep 2?

Reply:
The summary table is only printed out when solution invalid is found as coded in NonlinearSystem.C
![Screenshot 2023-02-12 at 9 37 44 PM](https://user-images.githubusercontent.com/118846840/218371817-9cfe6451-1441-4f57-a399-3eeb2f81ce42.png)